cmd/globalAlignment/globalAlignment.go

```diff
@@ -40,16 +40,16 @@ func cigarToGraph(target fasta.Fasta, query fasta.Fasta, aln []align.Cigar) *gen
 
 	//Creating the first node. This is done independent of all other number because this node has no 'previous' nodes.  All following code leverages the cigar output (second number printed in the Op position of the struct {}) to determine if the alignment returned a 0:match, 1:insertion, or 2:deletion. All indels are relative to the target sequence.
 	curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[0], 0)
-	genomeGraph.AddNode(answer, curr)
+	curr = genomeGraph.AddNode(answer, curr)
 	//Drawing the remaining nodes and all edges. Method for adding edges is based on previous nodes.
 	for i := 1; i < len(aln); i++ {
 		curr, targetEnd, queryEnd = genomeGraph.FaSeqToNode(target, query, targetEnd, queryEnd, aln[i], i)
-		genomeGraph.AddNode(answer, curr)
+		curr = genomeGraph.AddNode(answer, curr)
 		if aln[i].Op == 0 {
-			genomeGraph.AddEdge(answer.Nodes[i-1], curr, 1)
-			genomeGraph.AddEdge(answer.Nodes[i-2], curr, 0.5)
+			genomeGraph.AddEdge(&answer.Nodes[i-1], curr, 1)
```

Comment:
answer.Nodes has changed from a []*Node to a []Node.  This means that answer.Node[i-1] has changed from a *Node to a Node.  This change has happened, while AddEdge has stayed the same and wants a pointer to a Node instead of the node itself.  This means that we now have to put an "&" before while is code for "pointer to" or "memory address of".  AddEdge needs to keep taking pointers because each node should only have one copy in the program (that is held in Nodes) and all other places need to use pointers that point back to this position in Nodes.

Reply:
Ahhh. Yeah the change from []*Node to []Node made me think all pointers would disappear but makes sense we are keeping with pointing at nodes when constructing the edges. To be sure I understand and try to put it into my own words: if we didn't use pointers here, we would be copying the []Node into a new memory address when we construct the edges? And this would create a new copy of the node every time we add an edge?